crates/plugins/mediator-coordination/src/forward/routing.rs

```diff
@@ -0,0 +1,252 @@
+use std::sync::Arc;
+
+use axum::response::{IntoResponse, Response};
+
+use database::Repository;
+use didcomm::{AttachmentData, Message};
+use hyper::StatusCode;
+use mongodb::bson::doc;
+use serde_json::{json, Value};
+
+use crate::{
+    model::stateful::entity::{Connection, RoutedMessage},
+    web::{error::MediationError, AppState, AppStateRepository},
+};
+
+use super::error::RoutingError;
+
+/// Mediator receives forwarded messages, extract the next field in the message body, and the attachments in the message
+/// then stores the attachment with the next field as key for pickup
+pub async fn mediator_forward_process(
+    state: &AppState,
+    payload: Message,
+) -> Result<Message, Response> {
+  
+    let result = handler(state, payload).await.unwrap();
+    Ok(result)
+}
```

Comment:
I don't get the point of having such an empty function like this. The logic in the handler function should be placed here.

Reply:
the reason why i have a handler function is to handled errors which cannot be handled in this function due to its signature

Reply:
You just need to map the error to a `Response` type. It should be easy to do.

Reply:
Are you serious? A server must communicate all errors to the user. This why INTERNAL SERVER ERRORS exists. The client needs to be informed when something goes wrong, even if the issue is not directly related to their request.

Reply:
i agree with you internal server errors exits which is what will happen in this case, are you telling me we should give user logging errors about are mediator ?, example tell the user ```repository error```, because that is exposing our system which is even and error we do not want to handle as its fatal i prefere a panic with the user having his 500 statuscode.

Reply:
> i agree with you internal server errors exits which is what will happen in this case, are you telling me we should give user logging errors about are mediator ?, example tell the user `repository error`, because that is exposing our system which is even and error we do not want to handle as its fatal i prefere a panic with the user having his 500 statuscode.

I think i agree with these. Why should a user be receiving logging errors about the mediator?

Reply:
Why would you **"give users logging errors about our mediator"**? Logging internal details to the client is not advisable and not what I asked you to do. Instead, a generic error can be used to notify the client that their request was unsuccessful. The `map_err` is there to help you. Crashing should be reserved for unrecoverable errors, which is not the situation in this case. The server should not crash simply because a resource is inaccessible, that doesn't make sense. A server should remain responsive, even when the error is unrelated to the client, except in exceptional cases.

Reply:
check again the mediator is not crashing because it repository was not set, if this error is not unrecoverable then idk what it is

Reply:
The repository has already been set at this level. Your are just trying to access it. Please handle the error and go ahead.

Reply:
how can you be sure it was already been set, and if that be the case then we are sure they will be no error so no matter how it is handled it will never be reached and if you insist in handling that error which i think should'nt then you contradict the point you just said the repository is already set, are'nt you seeing it my way ?

Reply:
The repository is set here. At this point, the server has not been initiated, so it is safe to encounter a crash here and address the issue.
https://github.com/adorsys/didcomm-mediator-rs/blob/345b0c8fe5eef4f98fc9c96f6236d3d803cc64ba/crates/plugins/mediator-coordination/src/plugin.rs#L72-L109
In your case, you are accessing the repository to check for existing connections and store payloads. Various errors could occur, such as a temporary database issue and such. We typically respond by returning an error message to the client, advising them to try again later. If the error persists, our logs will alert us to the problem so we can resolve it. I believe this concludes the discussion on this matter. Please go ahead.

Reply:
i might be wrong, i think the last changes solves the issue